File: backend/app/routers/usage.py
from fastapi import APIRouter

from app.services.usage import UsageService
from app.utils.response import ResponseWrapper as R

router = APIRouter(prefix="/usage")


@router.get("/overview")
def usage_overview(
    start_at: str | None = None,
    end_at: str | None = None,
    task_id: str | None = None,
    provider_id: str | None = None,
    model_name: str | None = None,
    status: str | None = None,
):
    return R.success(
        UsageService.get_overview(
            start_at=start_at,
            end_at=end_at,
            task_id=task_id,
            provider_id=provider_id,
            model_name=model_name,
            status=status,
        )
    )


@router.get("/records")
def usage_records(
    start_at: str | None = None,
    end_at: str | None = None,
    task_id: str | None = None,
    provider_id: str | None = None,
    model_name: str | None = None,
    status: str | None = None,
):
    return R.success(
        UsageService.get_records(
            start_at=start_at,
            end_at=end_at,
            task_id=task_id,
            provider_id=provider_id,
            model_name=model_name,
            status=status,
        )
    )


@router.get("/task_summary")
def usage_task_summary(
    start_at: str | None = None,
    end_at: str | None = None,
    task_id: str | None = None,
    provider_id: str | None = None,
    model_name: str | None = None,
):
    return R.success(
        UsageService.get_task_summary(
            start_at=start_at,
            end_at=end_at,
            task_id=task_id,
            provider_id=provider_id,
            model_name=model_name,
        )
    )


@router.get("/task_calls/{task_id}")
def usage_task_calls(task_id: str):
    return R.success(UsageService.get_task_calls(task_id))

File: desktop/frontend/src/services/usage.ts
import request from '@/utils/request'

export interface UsageFilters {
  start_at?: string
  end_at?: string
  task_id?: string
  provider_id?: string
  model_name?: string
  status?: string
}

export interface UsageOverview {
  prompt_tokens: number
  completion_tokens: number
  total_tokens: number
  call_count: number
  task_count: number
}

export interface UsageRecord {
  id: number
  task_id: string
  provider_id: string
  provider_name: string
  model_name: string
  phase: string
  platform?: string
  video_id?: string
  video_title?: string
  prompt_tokens: number
  completion_tokens: number
  total_tokens: number
  status: string
  error_message?: string
  request_started_at?: string
  request_finished_at?: string
  duration_ms: number
  request_meta_json?: string
  created_at?: string
}

export interface TaskUsageSummary {
  task_id: string
  platform?: string
  video_id?: string
  video_title?: string
  call_count: number
  prompt_tokens: number
  completion_tokens: number
  total_tokens: number
  latest_call_at?: string
}

export const getUsageOverview = async (params: UsageFilters) => {
  return (await request.get('/usage/overview', { params })) as UsageOverview
}

export const getUsageRecords = async (params: UsageFilters) => {
  return (await request.get('/usage/records', { params })) as UsageRecord[]
}

export const getTaskUsageSummary = async (params: UsageFilters) => {
  return (await request.get('/usage/task_summary', { params })) as TaskUsageSummary[]
}

export const getTaskUsageCalls = async (taskId: string) => {
  return (await request.get(`/usage/task_calls/${taskId}`)) as UsageRecord[]
}
